cassiopeia/data/CassiopeiaTree.py

```diff
@@ -68,117 +73,291 @@ class CassiopeiaTree:
 
     def __init__(
         self,
-        character_matrix: pd.DataFrame,
+        character_matrix: Optional[pd.DataFrame] = None,
+        missing_state_indicator: int = -1,
         cell_meta: Optional[pd.DataFrame] = None,
         character_meta: Optional[pd.DataFrame] = None,
         priors: Optional[Dict[int, Dict[int, float]]] = None,
-        tree: Optional[Union[str, ete3.Tree, nx.DiGraph]] = None
+        tree: Optional[Union[str, ete3.Tree, nx.DiGraph]] = None,
     ):
 
-        self.character_matrix = character_matrix
+        self.__original_character_matrix = None
+        self.__current_character_matrix = None
+        if character_matrix is not None:
+            self.__original_character_matrix = character_matrix.copy()
+            self.__current_character_matrix = character_matrix.copy()
+        
+        self.missing_state_indicator = missing_state_indicator
         self.cell_meta = cell_meta
         self.character_meta = character_meta
         self.priors = priors
         self.__network = None
 
-        self.populate_tree(tree)
+        if tree is not None:
```

Comment:
If the tree already exists, do we want to automatically reset the branch lengths as well as the character states?
We could imagine this overwriting existing branch length data. What if instead we read in the existing branch lengths and then make the ages consistent

Reply:
What exactly do you mean by "If the tree already exists" @richardyz98 ? At this point the CassiopeiaTree is being initialized so there is no self.__network attribute. Also, the `tree` argument to `__init__` is also only used to pull out the topology, i.e. its branch lengths are ignored (we have decided to do this to have less responsibilities: if a user wants to set branch lengths of a CassiopeiaTree they should use the CassiopeiaTree API, we won't pull it out of their networkx or ete3 tree passed to `init`.)

Reply:
Ah I see. Perhaps we should make this explicit in the doc string, that the tree passed in should be a topology only and that branch lengths will be ignored on that tree